File: Python Warmup/Python Warmup/Student.py
class Student:
    def __init__(self,firstName = None ,lastName = None,id = None):
        if (not firstName ) and (not lastName) and (not id):
            self.firstName = ""
            self.lastName = ""
            self.id = 0
        else:
            self.firstName = firstName
            self.lastName = lastName
            self.id = id
        self.tests = []
        self.grade = ''

    def add_grade(self,score):
        self.tests.append(score)

    def compute_grade(self):
        sum = 0
        for score in self.tests:
            sum = sum + score
        avg = sum / len(self.tests)
        if avg > 90:
            grade = 'A'
        elif avg > 85:
            grade = 'A-'
        elif avg > 80:
            grade = 'B'
        else:
            grade = 'B'
        return grade

File: FirstAssignmentPart1/FirstAssignmentPart1/Student.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FirstAssignmentPart1
{
    class Student
    {
        string firstName;
        public string FirstName
        {
            get { return firstName; }
            set { firstName = value; }
        }
        string lastName;
        public string LastName
        {
            get { return lastName; }
            set { lastName = value; }
        }
        int id;
        public int Id
        {
            get { return id; }
            set { id = value; }
        }

        int test1;
        public int Test1
        {
            get { return test1; }
            set { test1 = value; }
        }

        int test2;
        public int Test2
        {
            get { return test2; }
            set { test2 = value; }
        }
        public string ComputeGrade()
        {
            string grade = "";
            double avg = 0.6 * test1 + 0.4 * test2;
            if (avg > 90)
            {
                grade = "A";
            }
            else if (avg > 85)
            {
                grade = "B";
            }
            else if (avg > 75)
            {
                grade = "C";
            }
            else if (avg > 65)
            {
                grade = "D";
            }
            else
            {
                grade = "F";

            }
            return grade;
        }
    }
}
